backend/api.py

```
@@ -379,6 +387,206 @@ def get_molecule_3d(index: int):
+        comparison = compare_molecules(smiles1, smiles2)
+        if comparison is None:
+            return jsonify({'success': False, 'error': 'Could not compare molecules'}), 400
+        
+        return jsonify({
+            'success': True,
+            **comparison
+        })
+    
+    except ValueError as e:
+        return jsonify({'success': False, 'error': str(e)}), 400
+    except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)